scripts/generate-iscsi-iqn

```diff
@@ -38,8 +38,8 @@ geniqn() {
 }
 
 # On upgrade, the host param will already be there - reuse it:
-if [ -n $(xe host-param-get uuid=${INSTALLATION_UUID} param-name=iscsi_iqn 2>/dev/null) ]; then
-    IQN=$(xe host-param-get uuid=${INSTALLATION_UUID} param-name=iscsi_iqn 2>/dev/null)
+if [ -n "$(xe host-param-get uuid=${INSTALLATION_UUID} param-name=iscsi_iqn 2>/dev/null)" ]; then
```

Comment:
Using `[[ -n $(...) ]]` may be better if variables used inside need quoting to prevent splitting.
Another option would be using a temporary variable like `_IQN` to separate collecting the output from the testing, it also avoids executing the command twice.

Reply:
I'll add the variable, because executing the xe command twice is indeed a bit silly.